### ISSUE TITLE
substrate-api-client is reporting a RuntimeMetadata:V12 even though I'm running it against a substrate-node with RuntimeMetadata:V14 in the source code.

### ISSUE THREAD

**Issue by ivanceras** [scs/substrate-api-client#238]
I expected this error due to version disparity.
```sh
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidMetadata(InvalidVersion)
```

But when I debug the `meta` variable, it shows that it is reporting V12 variant.

```sh
metadata: RuntimeMetadataPrefixed(1635018093, V12(OpaqueMetadata([24, 83, 121, 115, 116, 101, 109, 1, 24])))
```

So the current master branch of substrate-node-template uses substrate branch `polkadot-v0.19.20`.
I grep that code of substrate for "RuntimeMetadata" and only see `RuntimeMetadata::V14(metadata)` and not of the `V12` variant.

I also see that substrate-note-template is explicitly specifying `v14` feature in it's `Cargo.toml` config.
```
frame-metadata = { version = "15.0.0", features = ["v14"], package = "frame-metadata" }

```

I'm baffled by this issue, because I couldn't wrap my head of where the reported `V12` variant could have come from?

**Comment by ivanceras** [comment]
It turns out that if I put `--dev` as flag in running `substrate-node-template`, this error would not occur. Good thing I can now run the client against the substrate chain, but I'm still not quit sure if this is an intended behavior when `--dev` is used as a flag

**Comment by clangenb** [comment]
This is still very weird. Have you checked if there is a diff in the metadata, when you use the flag?

(We use the api-client on our end with different nodes without dev flags, and we never had an issue like this).

**Comment by ivanceras** [comment]
I think I've found the issue. Without the `--dev` flag the substrate chain is using the old database located at `<home>/.local/share/node-template/chains/local_testnet/db/full`. 
Using the `--dev` it creates a new database at a `/tmp` directory with a unique folder, so it is not reused in the next runs.

I issued a purge using `cargo run --release -- purge-chain` which removes the old data files from my previous runs. I'm suspecting the old database has the metadata stored there which at that time was at `RuntimeMetadata::V12`.

This would have been prevented also if I did all of this test in Dockerfile for ultimate reproducibility.

TLDR: issue a `cargo run --release -- purge-chain` to prevent this issue.
